packages/benchmark/src/benchmarkerator.js

```diff
@@ -425,43 +439,96 @@ const printBenchmarkStats = stats => {
+  const hc1 = `${'Counter'.padEnd(wc1)}`;
+  const dc1 = `${''.padEnd(wc1, '-')}`;
+
+  const wc2 = 6;
+  const hc2 = `${'Delta'.padStart(wc2)}`;
+  const dc2 = ` ${''.padStart(wc2 - 1, '-')}`;
+
+  const wc3 = 10;
+  const hc3 = `${'PerRound'.padStart(wc3)}`;
+  const dc3 = ` ${''.padStart(wc3 - 1, '-')}`;
+
+  log(``);
+  log(`${hc1} ${hc2} ${hc3}`);
+  log(`${dc1} ${dc2} ${dc3}`);
+  for (const [key, entry] of Object.entries(stats.counters)) {
+    const col1 = `${key.padEnd(wc1)}`;
+    const col2 = `${String(entry.delta).padStart(wc2)}`;
+    const col3 = `${pn(entry.deltaPerRound).padStart(wc3)}`;
     log(`${col1} ${col2} ${col3}`);
   }
+
+  const wg1 = 32;
+  const hg1 = `${'Gauge'.padEnd(wg1)}`;
+  const dg1 = `${''.padEnd(wg1, '-')}`;
+
+  const wg2 = 6;
+  const hg2 = `${'Start'.padStart(wg2)}`;
+  const dg2 = ` ${''.padStart(wg2 - 1, '-')}`;
+
+  const wg3 = 6;
+  const hg3 = `${'End'.padStart(wg3)}`;
+  const dg3 = ` ${''.padStart(wg3 - 1, '-')}`;
+
+  const wg4 = 6;
+  const hg4 = `${'Delta'.padStart(wg4)}`;
+  const dg4 = ` ${''.padStart(wg4 - 1, '-')}`;
+
+  log(``);
+  log(`${hg1} ${hg2} ${hg3} ${hg4}`);
+  log(`${dg1} ${dg2} ${dg3} ${dg4}`);
+  for (const [key, entry] of Object.entries(stats.gauges)) {
+    const col1 = `${key.padEnd(wg1)}`;
+    const col2 = `${String(entry.start).padStart(wg2)}`;
+    const col3 = `${String(entry.end).padStart(wg3)}`;
+    const col4 = `${String(entry.end - entry.start).padStart(wg4)}`;
+    log(`${col1} ${col2} ${col3} ${col4}`);
+  }
 };
 
-const organizeRoundsStats = (
-  rawBefore,
-  rawAfter,
-  elapsedTime,
-  cranks,
-  rounds,
-) => {
+const organizeRoundsStats = (rounds, perRoundStats) => {
+  let elapsedTime = 0;
+  let cranks = 0;
+  for (const { start, end } of perRoundStats) {
+    elapsedTime += Number(end.time - start.time);
+    cranks += end.cranks - start.cranks;
+  }
   const stats = {
     elapsedTime,
     cranks,
     rounds,
     timePerCrank: cranks ? elapsedTime / cranks : 0,
     timePerRound: elapsedTime / rounds,
     cranksPerRound: cranks / rounds,
-    data: {},
+    perRoundStats,
+    counters: {},
+    gauges: {},
   };
 
-  // Note: the following assumes rawBefore and rawAfter have the same keys.
-  for (const [key, value] of Object.entries(rawBefore)) {
+  // Note: the following assumes stats records all have the same keys.
+  const first = perRoundStats[0].start.rawStats;
+  const last = perRoundStats[perRoundStats.length - 1].end.rawStats;
+  for (const key of Object.keys(first)) {
     if (isMainKey(key)) {
-      const delta = rawAfter[key] - value;
-      stats.data[key] = {
-        delta,
-        deltaPerRound: delta / rounds,
-      };
+      if (Object.hasOwn(first, `${key}Max`)) {
+        stats.gauges[key] = {
+          start: first[key],
+          end: last[key],
+        };
+      } else {
+        let delta = 0;
+        for (const { start, end } of perRoundStats) {
+          delta += Number(end.rawStats[key] - start.rawStats[key]);
```

Comment:
the stats should already be numbers before math is performed (i.e. `getStats()` should return number fields with a number type)

Reply:
The `Number()` call here is not because stats aren't already numbers but because they are, more specifically, already BigInts.  I need the difference between them as a regular number so I can do a division that can have a non-integer result and JS doesn't allow you to mix BigInt and non-BigInt arithmetic without an explicit type conversion.  (I understand why TC39 did it this way, but that doesn't mean it's not super annoying.)